src/DotNetWorker.Core/Diagnostics/FunctionActivitySource.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.Functions.Worker.Diagnostics
+{
+    internal class FunctionActivitySourceFactory
+    {
+        private static readonly ActivitySource _activitySource = new(TraceConstants.FunctionsActivitySource, TraceConstants.FunctionsActivitySourceVersion);
+        private readonly string _schemaVersionUrl;
+        private readonly Lazy<IReadOnlyDictionary<string, string>> _attributeMap;
+
+        public FunctionActivitySourceFactory(IOptions<WorkerOptions> options)
+        {
+            _attributeMap = new Lazy<IReadOnlyDictionary<string, string>>(() => GetMapping(options.Value.OpenTelemetrySchemaVersion));
+            _schemaVersionUrl = TraceConstants.OpenTelemetrySchemaMap[options.Value.OpenTelemetrySchemaVersion];
+        }
+
+        public Activity? StartInvoke(FunctionContext context)
+        {
+            var activity = _activitySource.StartActivity(TraceConstants.FunctionsInvokeActivityName, ActivityKind.Internal, context.TraceContext.TraceParent,
```

Comment:
There is a `StartActivity` overload where you can pass in `ActivityContext parentContext`, which is a struct to hold traceParent and traceState.

``` CSharp
ActivityContext.TryParse(context.TraceContext.TraceParent, context.TraceContext.TraceState, out ActivityContext parentContext);
```

Reply:
Wouldn't we have to do this first, though, even if there's nothing Listening? I wanted to only do stuff like this if we actually create an Activity. Or maybe there's some other benefit to using the ActivityContext?

Reply:
Oh we could wrap all this in `_activitySource.HasListeners()`...